README.md

```diff
@@ -254,48 +254,40 @@ sections:
-| $sort   | string | 'asc'      | Comments are ordered by timestamp; 'desc' lists latest first, 'asc' lists chronologically                                                          |
+| Name      | Type   | Default    | Description                                                                                                                                            |
+|-----------|--------|------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| $query    | string | 'approved' | Selects the comments to be included in the structure; possible values are either a valid status('approved', 'pending', 'all') or a single comment UID. |
+| $language | string | null       | `null` for all languages or a two-letter language code (e.g. `en`); comments saved with no language information are returned for all languages         |
 
 #### Return
 
-* When requesting with a status: array with all comments for the requested object (see "Data structure" chapter below for details); empty array if none. A field `'pageid'` is added to each comment's array, indicating the Kirby page ID the comment belongs to.
-* When requesting with a UID: array with the data of the comment; boolean `false` if UID is not valid.
+Structure with all comments for the requested object that meet the criteria; empty structure if no matches.
 
-The following example shows the most minimal comment and webmentions possible (displayed fields are compulsory):
+#### Usage
+
+When looping through the returned structure object, the comment data can be retrieved in various ways:
 
 ```
-Array (
-    [0] => Array
-        (
-            [status] => approved
-            [text] => Most minimal comment possible
-            [timestamp] => 2020-04-01 11:30
-            [type] => comment
-            [uid] => 1m6los473p
-            [pageid] => notes/exploring-the-universe
-        )
-    [1] => Array
-        (
-            [source] => https://example.com
-            [status] => approved
-            [timestamp] => 2020-04-01 11:32
-            [type] => like
-            [uid] => 126los473p
-            [pageid] => notes/exploring-the-universe
-        )
-)
+foreach ( $testpage->commentions('all') as $item ) {
+    print_r($item->content());
+}
 ```
 
-The field `pageid` does not come from the data storage; it is added by the parser to the output of the page(s) method by default.
+`$item->content()` returns a content object with all the fields stored for the comment; each field within can also be accessed by its name, e.g. `$item->name()` etc. The HTML stored in field `$item->text()` has been cleaned up with HTML Purifier; the originally submitted HTML (NB. This may contain malicious code, never use without appropriate filtering) is stored in field `$item->text_unsafe()`.
```

Comment:
Currently, the `$item->text()` method returns sanitized HTML and `$item->content()->get('text')` returns the raw, unfiltered value. Not ideal, but much easier to handle when calling the `$item->toArray()` method in the backend.

Reply:
Would you be able to check one more time that I've got these right?

https://github.com/sebastiangreger/kirby3-commentions/blob/40-readme/README.md#page-commentions

...not sure about the difference between `$item->toArray()` and `$item->content()->toArray()` right now. Thx!

Reply:
@sebastiangreger It boils down to this:

- `$item->toArray()` We control, what’s included in the array. This currently includes - `name_formatted` and `source_formatted`
`$item->content()->toArray()` Kirby gives us the raw, unformatted content as array. As we don’t have a custom `Content` object, we just get what we fed it with.

This is not ideal and probably hard to understand. I'm still unsure, how to handle this in an elegant way. `$commention->text()` behaves kind of strange, because it pretends to return a value, that has been altered. AFAIK, this is not done anywhere else in Kirby and it’s kinf of intransparent.

However, I like the idea of having processed values such as `name_formatted` in the array export, as they make the API implementation much easier and are probably very usefull for most users anyway. Maybe we should consider to let `$commention->text()` return the raw value and use either `$commention->textSanitized()` or something like `$commention->text()->toSanitizedHTML()` to make it more transparent, what happens under the hood? Let’s discuss it in our phone call later this week.